src/Vulkan/VkRHI.cpp

```diff
@@ -275,24 +288,44 @@ UniqueHandle<RHIGraphicsPipelineState> VkRHI::CreateGraphicsPipelineState(const
 
 UniqueHandle<RHIComputePipelineState> VkRHI::CreateComputePipelineState(const ComputePipelineStateKey& key)
 {
-    return MakeUnique<VkComputePipelineState>(key);
+    return MakeUnique<VkComputePipelineState>(key, *device, *PSOCache);
 }
 
-UniqueHandle<RHIResourceLayout> VkRHI::CreateResourceLayout(const FeatureChecker& featureChecker)
+UniqueHandle<RHIResourceLayout> VkRHI::CreateResourceLayout(const FeatureChecker& featureChecker,
+                                                            RHIDescriptorPool& descriptorPool)
 {
-    VEX_NOT_YET_IMPLEMENTED();
-    return MakeUnique<VkResourceLayout>(/*device, reinterpret_cast<const VkFeatureChecker&>(featureChecker)*/);
+    return MakeUnique<VkResourceLayout>(*device,
+                                        reinterpret_cast<const VkDescriptorPool&>(descriptorPool),
+                                        reinterpret_cast<const VkFeatureChecker&>(featureChecker));
 }
 
 UniqueHandle<RHITexture> VkRHI::CreateTexture(const TextureDescription& description)
 {
-    VEX_NOT_YET_IMPLEMENTED();
-    return MakeUnique<VkTexture>(GetGPUContext(), TextureDescription(description));
+    UniqueHandle<VkImageTexture> texture = MakeUnique<VkImageTexture>(GetGPUContext(), TextureDescription(description));
+
+    auto commandPool = CreateCommandPool();
+    auto& vkCmdList = reinterpret_cast<VkCommandList&>(*commandPool->CreateCommandList(CommandQueueTypes::Compute));
+    vkCmdList.Open();
+
+    vkCmdList.Transition(*texture, RHITextureState::UnorderedAccess);
+
+    vkCmdList.Close();
+
+    const auto& graphicsQueue = commandQueues[CommandQueueTypes::Compute];
+
+    ::vk::SubmitInfo submitInfo{ .commandBufferCount = 1, .pCommandBuffers = &*vkCmdList.commandBuffer };
+    VEX_VK_CHECK << graphicsQueue.queue.submit(1, &submitInfo, nullptr);
+
+    // This shouldn't be needed i think
+    // TODO: Find a way to make sure this isn't needed. It screams otherwise because the commandlist is supposedly in
+    // use
+    VEX_VK_CHECK << graphicsQueue.queue.waitIdle();
+
```

Comment:
Oh wait what? We probably don't want to open and submit an entire command list just for a texture's creation, is it not possible to specify an initial state in Vk? If not, I'd just leave it in the default state and only transition when needed (eg: for UAV state, only when the dispatch is called in command context).